### PR TITLE
Kubernetes documentation enhancements

### PR DESCRIPTION
**Pull request by greenaar**
Updated volume mount example to include both plugin and non-plugin based, as `settings` is incompatible with `commands`

Added a link to the Kubernetes PVC documentation, and a note about ReadWriteMany.

Added details around setting up prometheus monitoring where both prometheus and woodpecker are in the same cluster.

Included troubleshooting for prometheus failing to scrape metrics.

Added links and examples for serviceAccountName configuration.

### CODE REVIEW

File: docs/docs/30-administration/22-backends/40-kubernetes.md [hunk 1]
```diff
@@ -133,6 +160,21 @@ steps:
...
+  - name: "Edit cache"
+    image: alpine:latest
+    volumes:
+      - woodpecker-cache:/woodpecker/src/cache
+    commands:
+      - echo "Hello World" > /woodpecker/src/cache/output.txt
+    [...]
+```
+
+Please note that `settings:` is reserved for use with plugins, and is incompatible with `commands:`
```
Comment (zc-devs) [inline docs/docs/30-administration/22-backends/40-kubernetes.md:176]: This is unnecessary here.

File: docs/docs/30-administration/22-backends/40-kubernetes.md [hunk 2]
```diff
@@ -308,3 +350,40 @@ Determines if containers must be required to run as non-root users.
 > Default: empty
 
 Secret names to pull images from private repositories. See, how to [Pull an Image from a Private Registry](https://kubernetes.io/docs/tasks/configure-pod-container/pull-image-private-registry/).
+
+## Metrics
+
```
Comment (zc-devs) [inline docs/docs/30-administration/22-backends/40-kubernetes.md:355]: And this. Move in deployment instruction or in helm chart repository.
https://woodpecker-ci.org/docs/administration/deployment-methods/kubernetes

File: docs/docs/30-administration/22-backends/40-kubernetes.md (hunk 1 above)
Comment (zc-devs) [reply docs/docs/30-administration/22-backends/40-kubernetes.md]: https://woodpecker-ci.org/docs/usage/workflow-syntax#commands

File: docs/docs/30-administration/22-backends/40-kubernetes.md [hunk 3]
```diff
@@ -50,6 +50,21 @@ See the [Kubernetes documentation](https://kubernetes.io/docs/concepts/container
...
 
+```yaml
+steps:
+  - name: 'My kubernetes step'
+    image: alpine
+    commands:
+      - echo "Hello world"
+    backend_options:
+      kubernetes:
+        # Use the service account `default` in the current current namespace.
```
Comment (zc-devs) [inline docs/docs/30-administration/22-backends/40-kubernetes.md:61]:
```suggestion
        # Use the service account `default` in the current namespace.
```

**Review by zc-devs — APPROVED** [review]

File: docs/docs/30-administration/22-backends/40-kubernetes.md [hunk 4]
```diff
@@ -119,7 +134,19 @@ steps:
 ### Volumes
 
 To mount volumes a PersistentVolume (PV) and PersistentVolumeClaim (PVC) are needed on the cluster which can be referenced in steps via the `volumes` option.
-Assuming a PVC named `woodpecker-cache` exists, it can be referenced as follows in a step:
+
+Persistent volumes need to be manually generated, use the Kubernetes [Persistent Volumes](https://kubernetes.io/docs/concepts/storage/persistent-volumes/) documentation as a reference.
+
+_If your PVC is not highly available or NFS based, you may need to also integrate affinity settings to ensure your steps run on the correct node._
+
+NOTE: If you plan to use this volume on more than one workflow concurrently, make sure you have configured the PVC with:
```
Comment (xoxys) [inline docs/docs/30-administration/22-backends/40-kubernetes.md:142]:
```suggestion
NOTE: If you plan to use this volume in more than one workflow concurrently, make sure you have configured the PVC in `RWX` mode. Keep in mind that this feature must be supported by the used CSI driver:
```

File: docs/docs/30-administration/22-backends/40-kubernetes.md [hunk 5]
```diff
@@ -50,6 +50,21 @@ See the [Kubernetes documentation](https://kubernetes.io/docs/concepts/container
...
+    commands:
+      - echo "Hello world"
+    backend_options:
+      kubernetes:
+        # Use the service account `default` in the current namespace.
+        # This usually the same as wherever woodpecker is deployed.
+        serviceAccountName: default
+```
+
+To give steps access to the kubernetes api via service account, look at [RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)
```
Comment (xoxys) [inline docs/docs/30-administration/22-backends/40-kubernetes.md:66]:
```suggestion
To give steps access to the Kubernetes API via service account, take a look at [RBAC Authorization](https://kubernetes.io/docs/reference/access-authn-authz/rbac/)
```

**Review by xoxys — COMMENTED** [review]
Thanks for expanding the documents, and sorry for being so picky 😇

**Review by xoxys — APPROVED** [review]

**Comment by 6543** [comment]
CI failed ...

**Pull request merged**